### DISCUSSION TITLE
Filter web part search box behavior

### DISCUSSION THREAD

**Discussion by ghost** [microsoft-search/pnp-modern-search#1828] [Q&A]
What is the threshold for the search box to appear in the filter web part for filter fields?
It seems that the search box appears for (almost) every filter field? Even for the one with very limited values where it is not needed (and takes up unnecessary space).

Does anyone know this?
Thank you!

**Comment by wobba** [comment]
Can you explain a bit more what you are asking? Don't quite understand.

**Reply by ghost** [reply]
In v4.5.3 this feature has been added: "Added a search box and a scroll behavior to handle long list of filter values."

So, my question is: what is considered to be a "**long** list of filter values"? As the search box also shows for short lists of filter values (10 or less) in my environment.

**Reply by wobba** [reply]
"long" means outside of what the screen can show - it will scale and not break the UX allowing you to pick any item.

**Reply by LetMeSP** [reply]
I have a similar issue. What is the expected behavior if I have 10 available filters but decided to show only 6 (4 are hidden). Should the filter search include the 4 hidden ones? In my testing it is only searching among the visible ones...

**Reply by wobba** [reply]
Screenshots and config examples would help to understand the issue :)

**Reply by LetMeSP** [reply]
Here are the screenshots:

If I show all filter options I can search all that are visible:
![2022-02-28_14-04-26](https://user-images.githubusercontent.com/41652178/156066683-23f4283e-f67e-4a13-ab5f-a0b5fb160572.png)

If I limit the number of values the filter options not currently shown are not available when searching. Is this the expected behavior? 
![2022-02-28_14-03-27](https://user-images.githubusercontent.com/41652178/156066825-b9caf87b-57cc-4f67-90af-90cff5d29702.png)

**Reply by FranckyC** [reply]
@LetMeSP the number of values for a filter determines the number of refiner values retrieved by the search API so this is the expected behavior in your case. This setting has no link with the filter search box and is normally used when you must display a large number of values beyond the default limit (50 if I remember). The number of filters you see depends on the height of the control, nothing more. You can set this value as wanted by modifying the template:

![image](https://user-images.githubusercontent.com/5659122/156085120-9a6df74a-9a1f-404b-9cd0-1981d939f656.png)

**Reply by LetMeSP** [reply]
@FranckyC - thank you for the clarification!
